README.md

```diff
@@ -148,86 +168,166 @@ Klaviyo.createEvent(event)
-    super.onCreate(savedInstanceState)
-    
     /* ... */
-    
-    // Initialize is required to use any Klaviyo SDK functionality 
-    Klaviyo.initialize("KLAVIYO_PUBLIC_API_KEY", applicationContext)
 
     // Fetches the current push token and registers with Push SDK
     FirebaseMessaging.getInstance().token.addOnSuccessListener { pushToken ->
         Klaviyo.setPushToken(pushToken)
     }
+}
+```
+
+**Reminder**: `Klaviyo.initialize` is required before using any other Klaviyo SDK functionality, even 
+if you are only using the SDK for push notifications and not analytics.
+
+#### Push tokens and multiple profiles
+Klaviyo SDK will disassociate the device push token from the current profile whenever it is reset by calling 
+`setProfile` or `resetProfile`. You should call `setPushToken` again after resetting the currently tracked profile
+to explicitly associate the device token to the new profile.
+
+### Receiving Push Notifications
+`KlaviyoPushService` will handle displaying all notifications via the `onMessageReceived` method regardless of
+whether the app is in the foreground or background. If you wish to customize how notifications are displayed, see
+[Advanced Setup](#advanced-setup).
+
+#### Rich Push
+**Rich Push** is the ability to add images to your push notification messages. This feature is supported in 
+version 1.3.1 and up of the Klaviyo Android SDK. No additional setup is needed to support rich push. 
+Downloading the image and attaching it to the notification is handled within `KlaviyoPushService`. If an image
+fails to download (e.g. if the device has a poor network connection) the notification will be displayed without
+an image after the download times out.
+
+#### Tracking Open Events
+To track push notification opens, you must call `Klaviyo.handlePush(intent)` when your app is launched from an intent.
+This method will check if the app was opened from a notification originating from Klaviyo and if so, create an 
+`Opened Push` event with required message tracking parameters. For example:
+
+```kotlin
+override fun onCreate(savedInstanceState: Bundle?) {
+    /* ... */
 
     onNewIntent(intent)
 }
 
 override fun onNewIntent(intent: Intent?) {
-    super.onNewIntent(intent)
+    /* ... */
 
     // Tracks when a system tray notification is opened
     Klaviyo.handlePush(intent)
 }
 ```
 
-To specify a notification icon, add the following metadata to your app manifest.
-Absent this, the application's launcher icon will be used.
+**Note** intent handling may differ depending on your app's architecture. Adjust this example to your use-case, 
+ensuring that `Klaviyo.handlePush(intent)` is called when your app is opened from a notification.
+
+#### Deep Linking 
+**Deep Links** allow you to navigate to a particular page within your app in response to the user 
+opening a notification. There are broadly three steps to implement deep links in your app. 
+
+##### Add intent filters for incoming links
```

Comment:
Rather than subheadering for each step, should we have one subheader for `Deep Linking via URI Schemes` for example and including the steps to get set up with this form of deep linking. And if we want to mention support or lack-thereof of other link types we can include additional subheaders for those

Reply:
Hm maybe making these H5s is causing confusion, it is meant to be like 3 steps required to set up deep links. I think Ajay was mirroring these three steps from the google documentation about adding deep links to an app.

Reply:
Ya I understood that, I just thought maybe we dont want each step to be a header and just have deep linking headered by type. Going from H2 to H5 in section nesting seems to be getting pretty deep too is my only other thought

Reply:
Makes sense, I will switch back to numbered steps